pytorch_lightning/accelerators/accelerator.py

```diff
@@ -0,0 +1,365 @@
+                    :paramref:`~pytorch_lightning.trainer.trainer.Trainer.truncated_bptt_steps` > 0.
+
+        """
+        batch = self.to_device(args[0])
+
+        args[0] = batch
+
+        with self.precision_plugin.train_step_context():
+            with self.training_type_plugin.train_step_context():
+                return self.lightning_module.training_step(*args)
+
+    def validation_step(self, args):
+        """The actual validation step.
+
+        Args:
+            args: the arguments for the models validation step. Can consist of the following:
+                batch (:class:`~torch.Tensor` | (:class:`~torch.Tensor`, ...) | [:class:`~torch.Tensor`, ...]):
+                    The output of your :class:`~torch.utils.data.DataLoader`. A tensor, tuple or list.
+                batch_idx (int): The index of this batch
+                dataloader_idx (int): The index of the dataloader that produced this batch
+                    (only if multiple val dataloaders used)
+        """
+        batch = self.to_device(args[0])
+
+        args[0] = batch
+
+        with self.precision_plugin.val_step_context():
+            with self.training_type_plugin.val_step_context():
+                return self.lightning_module.validation_step(*args)
+
+    def test_step(self, args):
+        """The actual test step.
+
+        Args:
+            args: the arguments for the models test step. Can consist of the following:
+                batch (:class:`~torch.Tensor` | (:class:`~torch.Tensor`, ...) | [:class:`~torch.Tensor`, ...]):
+                    The output of your :class:`~torch.utils.data.DataLoader`. A tensor, tuple or list.
+                batch_idx (int): The index of this batch.
+                dataloader_idx (int): The index of the dataloader that produced this batch
+                    (only if multiple test dataloaders used).
+        """
+        batch = self.to_device(args[0])
+
+        args[0] = batch
+
+        with self.precision_plugin.test_step_context():
+            with self.training_type_plugin.test_step_context():
+                return self.lightning_module.test_step(*args)
+
+    def training_step_end(self, output):
+        """A hook to do something at the end of the training step
+
+        Args:
+            output: the output of the training step
+        """
+        return output
+
+    def test_step_end(self, output):
+        """A hook to do something at the end of the test step
+
+        Args:
+            output: the output of the test step
+        """
+        return output
+
+    def validation_step_end(self, output):
+        """A hook to do something at the end of the validation step
+
+        Args:
+            output: the output of the validation step
+        """
+        return output
+
+    def process_dataloader(
+        self, dataloader: Union[Iterable, torch.utils.data.DataLoader]
+    ) -> Union[Iterable, torch.utils.data.DataLoader]:
+        """Wraps the dataloader if necessary
```

Comment:
what does the wrapping do? where is it unwrapped?

Reply:
It is never unwrapped. Just for some kinds of training (like tpu for example) the dataloader has to be wrapped.